indico/modules/events/abstracts/controllers/reviewing.py

```diff
@@ -51,8 +58,97 @@ class RHAbstractsDownloadAttachment(RHAbstractsReviewBase):
     }
 
     def _checkParams(self, params):
-        RHAbstractsReviewBase._checkParams(self, params)
+        RHAbstractReviewBase._checkParams(self, params)
         self.abstract_file = AbstractFile.get_one(request.view_args['file_id'])
 
     def _process(self):
         return self.abstract_file.send()
+
+
+def get_user_review_for_track(user, abstract, track):
+    results = [review for review in abstract.get_track_reviews(track) if review.user == user]
+    assert len(results) <= 1
+    return results[0] if results else None
+
+
+def build_review_form(abstract, track):
+    review_form_class = make_review_form(abstract.event_new)
+    review_for_track = get_user_review_for_track(session.user, abstract, track)
+
+    if review_for_track:
+        answers = {'question_{}'.format(rating.question.id): rating.value
+                   for rating in review_for_track.ratings}
+        defaults = FormDefaults(obj=review_for_track, **answers)
+    else:
+        defaults = FormDefaults()
+
+    return review_form_class(prefix="track-{}".format(track.id), obj=defaults)
+
+
+class AbstractPageMixin(AbstractMixin):
+    """Display abstract management page"""
+
+    def _process(self):
+        review_forms = {track.id: build_review_form(self.abstract, track)
+                        for track in self.abstract.reviewed_for_tracks
+                        if track.is_user_reviewer(session.user)}
+        judgement_form = AbstractJudgmentForm(abstract=self.abstract)
+
+        return WPManageAbstracts.render_template('abstract.html', self._conf, abstract=self.abstract,
+                                                 judgement_form=judgement_form, review_forms=review_forms,
+                                                 management=self.management)
+
+
+class RHJudgeAbstract(RHAbstractReviewBase):
```

Comment:
OTOH, why are we extending from `RHAbstractReviewBase`? IIRC, we agreed judges wouldn't be able to review unless listed as such.